nodes/persistence-node/src/services/IpfsGatewayApi.ts

```diff
@@ -78,33 +91,51 @@ export class IpfsGatewayApi {
 
       const buffer = Buffer.from(data);
 
+      res.set('Content-disposition', 'attachment; filename=' + filename);
+
       res.send(buffer);
     }));
 
     app.get('/api/v0/resolve', handleError(async (req, res) => {
       const hash = req.query.arg as string;
 
-      const resolvedPath = await ipfs.resolve(`/ipfs/${hash}`); 
+      const resolvedPath = await ipfs.resolve(`/ipfs/${hash}`);
 
       res.json({
         path: resolvedPath
       });
     }));
 
-    app.post('/add', upload.fields([ { name: "files"}, { name: "options", maxCount: 1 } ]), handleError(async (req, res) => {
-      if(!req.files) {
+    app.get('/api/v0/ls/:hash', handleError(async (req, res) => {
```

Comment:
IPFS does have an endpoint like this https://docs.ipfs.io/reference/http/api/#api-v0-ls
But it does something different
Rename this endpoint to '/ipfs/:hash' to mimick ipfs.io/ipfs/:hash